modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java

```diff
@@ -137,14 +140,25 @@
 
     OverviewsController overviewsController;
 
+    private CoordinateReferenceSystem granuleCRS;
+
+    public GeneralEnvelope coverageEnvelope;
```

Comment:
I'll change the name to be more accurate and take a look at the visibility to make sure it's correct.